src/RadhydroSimulation.hpp

```diff
@@ -469,152 +474,411 @@ void RadhydroSimulation<problem_t>::advanceHydroAtLevel(int lev, amrex::Real tim
 		fluxScaleFactor = 1.0;
 	}
 
-	// update ghost zones [old timestep]
-	fillBoundaryConditions(state_old_[lev], state_old_[lev], lev, time);
+  std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+
+  amrex::MFIter::allowMultipleMFIters(true);
+
+  #pragma omp parallel sections
```

Comment:
Is there a good reason to use sections? I think this can be replaced with `#pragma omp parallel num_threads(2)`, and then within this construct, we can use an if statement to check the thread ID with `omp_get_thread_num()`.

This will enable us to use MPI_THREAD_FUNNELED (which has better performance) instead of MPI_THREAD_MULTIPLE. (See https://cvw.cac.cornell.edu/hybrid/overlapcommwork)